luskan/src/test/java/org/opencadc/luskan/NumCopiesConverterTest.java

```diff
@@ -0,0 +1,170 @@
+ *  OpenCADC project.                    OpenCADC.
+ *
+ *  OpenCADC is free software:           OpenCADC est un logiciel libre ;
+ *  you can redistribute it and/or       vous pouvez le redistribuer ou le
+ *  modify it under the terms of         modifier suivant les termes de
+ *  the GNU Affero General Public        la “GNU Affero General Public
+ *  License as published by the          License” telle que publiée
+ *  Free Software Foundation,            par la Free Software Foundation
+ *  either version 3 of the              : soit la version 3 de cette
+ *  License, or (at your option)         licence, soit (à votre gré)
+ *  any later version.                   toute version ultérieure.
+ *
+ *  OpenCADC is distributed in the       OpenCADC est distribué
+ *  hope that it will be useful,         dans l’espoir qu’il vous
+ *  but WITHOUT ANY WARRANTY;            sera utile, mais SANS AUCUNE
+ *  without even the implied             GARANTIE : sans même la garantie
+ *  warranty of MERCHANTABILITY          implicite de COMMERCIALISABILITÉ
+ *  or FITNESS FOR A PARTICULAR          ni d’ADÉQUATION À UN OBJECTIF
+ *  PURPOSE.  See the GNU Affero         PARTICULIER. Consultez la Licence
+ *  General Public License for           Générale Publique GNU Affero
+ *  more details.                        pour plus de détails.
+ *
+ *  You should have received             Vous devriez avoir reçu une
+ *  a copy of the GNU Affero             copie de la Licence Générale
+ *  General Public License along         Publique GNU Affero avec
+ *  with OpenCADC.  If not, see          OpenCADC ; si ce n’est
+ *  <http://www.gnu.org/licenses/>.      pas le cas, consultez :
+ *                                       <http://www.gnu.org/licenses/>.
+ *
+ *  : 5 $
+ *
+ ************************************************************************
+ */
+
+package org.opencadc.luskan;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.log4j.Level;
+import org.apache.log4j.Logger;
+import org.junit.Assert;
+import org.junit.Test;
+
+import ca.nrc.cadc.tap.TapQuery;
+import ca.nrc.cadc.tap.schema.TapSchema;
+import ca.nrc.cadc.util.Log4jInit;
+import ca.nrc.cadc.util.MultiValuedProperties;
+import ca.nrc.cadc.uws.Parameter;
+
+public class NumCopiesConverterTest {
+    private static final Logger log = Logger.getLogger(NumCopiesConverterTest.class);
+
+    static {
+        Log4jInit.setLevel("org.opencadc.luskan", Level.INFO);
+        Log4jInit.setLevel("net.sf.jsqlparser", Level.INFO);
+    }
+
+    private static final TapSchema tapSchema = TestUtil.mockTapSchema();
+
+    @Test
+    public void testSelect() {
+        String query = "select *, num_copies() from inventory.Artifact where id = '{some artifact id}'";
+        String expected = "SELECT inventory.Artifact.id, inventory.Artifact.contentLength, cardinality(inventory.artifact.sitelocations) FROM inventory.Artifact WHERE id = '{some artifact id}'";
```

Comment:
The AllColumnsConverter (for *) clearly preserves the case of the table name from the tap_schema, so you should expect `num_copies(inventory.Artifact.siteLocations)` here... and fix code to make that work. 

Well, there won't be a fix needed because the test in `doTest` does a toLowercase (which is common in this scenario - it gets you past a lot of needless discrepancies)... I'd probably add an additional arg to doTest like 
`String expectedCardinalityFunc` 
and do an 
`Assert.assertTrue(sql.contains(expectedCardinalityFunc))` 
before the toLowerCase

Reply:
Updated the tests to check for the camel case argument to cardinality before doing the ignore case check of the SQL.